DSP_Plugins.GalacticScale/Scripts/PatchPlanetSize/PatchOnUIBuildingGrid.cs

```diff
@@ -0,0 +1,61 @@
+﻿using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Patch = GalacticScale.Scripts.PatchPlanetSize.PatchForPlanetSize;
+
+namespace GalacticScale.Scripts.PatchPlanetSize {
+    [HarmonyPatch(typeof(UIBuildingGrid))]
+    public class PatchUIBuildingGrid {
+
+        public static int refreshGridRadius = -1;
+
+        //segment count to 512 lut
+        public static Dictionary<int, int[]> LUT512 = new Dictionary<int, int[]>();
+
+        [HarmonyPrefix]
+        [HarmonyPatch("Update")]
+        public static bool Update(UIBuildingGrid __instance, Material ___material, Material ___altMaterial) {
+            if (Patch.EnableResizingFeature.Value || Patch.EnableLimitedResizingFeature.Value) {
+                int segments;
+                if (refreshGridRadius != -1) {
+                    if (___material == null) {
+                        Patch.Debug("Material was null!", BepInEx.Logging.LogLevel.Debug, true);
+                        return false;
+                    }
+                    segments = (int) (refreshGridRadius / 4f + 0.1f) * 4;
+                    if (LUT512.ContainsKey(segments)) {
+                        Patch.Debug("Updating LUT for radius + " + refreshGridRadius + " and segments " + segments + "!", BepInEx.Logging.LogLevel.Debug, true);
+                        UpdateTextureToLUT(___material, segments);
+                    }
+                    else {
+                        //TODO
+                        Patch.Debug("LUT512 did not yet contain the texture for refreshing.", BepInEx.Logging.LogLevel.Debug, true);
+                        refreshGridRadius = -1;
+                    }
+                }
+                return false;
```

Comment:
This must always continue running. I only piggybacked on Update because I have the relevant info here and it gets called regularly. But the regular Update method must still run regardless.